walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,210 @@
+    List<TimeRange> mergeSchedule = (preExistingTimeRanges != null)? preExistingTimeRanges: new ArrayList<>();
+
+    for (String attandee: allSchedules.keySet()) {
+        ArrayList<TimeRange> attandeeSchedule = allSchedules.get(attandee);
+        Comparator<TimeRange> comparatorByStart = TimeRange.ORDER_BY_START;
+        Collections.sort(attandeeSchedule, comparatorByStart);
+        allSchedules.put(attandee, attandeeSchedule);
+        
+        System.out.println(attandeeSchedule); 
+        int index1 = 0;
+        int index2 = 0;
+        ArrayList<TimeRange> newMergeSchedule = new ArrayList<>();
+        while (index1 < mergeSchedule.size() && index2 < attandeeSchedule.size()) {
+            TimeRange mergeSlot = mergeSchedule.get(index1);
+            TimeRange attandeeSlot = attandeeSchedule.get(index2);
+
+            if (mergeSlot.overlaps(attandeeSlot)) {
+                TimeRange newMergeSlot = mergeTwoOverlappedTimeRanges(mergeSlot, attandeeSlot);
+                newMergeSchedule = mergeTimeRangeWithAccumulatedTimeRanges(newMergeSchedule, newMergeSlot);
+                index1++;
+                index2++;
+            }
+            else {
+                if (mergeSlot.start() < attandeeSlot.start()) {
+                    newMergeSchedule = mergeTimeRangeWithAccumulatedTimeRanges(newMergeSchedule, mergeSlot);
+                    index1++;
+                } else {
+                    newMergeSchedule = mergeTimeRangeWithAccumulatedTimeRanges(newMergeSchedule, attandeeSlot);
+                    index2++;
+                }
+            }
+
+        }
+        while (index1 < mergeSchedule.size()) {
+            TimeRange mergeSlot = mergeSchedule.get(index1);
+            newMergeSchedule = mergeTimeRangeWithAccumulatedTimeRanges(newMergeSchedule, mergeSlot);
+            index1++;
+        }
+        while (index2 < attandeeSchedule.size()) {
+            TimeRange attandeeSlot = attandeeSchedule.get(index2);
+            newMergeSchedule = mergeTimeRangeWithAccumulatedTimeRanges(newMergeSchedule, attandeeSlot);
+            index2++;
+        }
+        mergeSchedule = newMergeSchedule;
+    }
+    return mergeSchedule;
+  }
+  /* Consider if each 'missing' TimeRange in the merged schedule will be long enough to fit the meeting in
+  param: @mergeSchedule: a list of TimeRange that merges all schedule
+         @request: a MeetingRequest contains the duration of the meeting
+  return: return a list possible TimeRange that fits all given schedules
+  */
+  public static List<TimeRange> fitAllSchedules(List<TimeRange> mergeSchedule, MeetingRequest request) {
+    ArrayList<TimeRange> possibleSlots = new ArrayList<>();
+
+    if (mergeSchedule.size() == 0) {
+        if (TimeRange.END_OF_DAY - TimeRange.START_OF_DAY >= request.getDuration()) {
+            possibleSlots.add(TimeRange.WHOLE_DAY);
+        }
+    }
+    else {
+        for (int i = -1; i < mergeSchedule.size(); i++) {
```

Comment:
Is there a reason why the index is set to -1 instead of the usual 0? It seems that the if statement below can be changed to:

```
if (i == 0) {
  TimeRange next = mergeSchedule.get(i);
  if (next.start() - TimeRange.START_OF_DAY >= request.getDuration()) {
    possibleSlots.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, next.start(), false));
  }
}
```

Maybe I'm missing something? Would be good if you can clarify.